src/targets/gpu/gemm_impl.cpp

```diff
@@ -27,6 +28,22 @@ rocblas_datatype get_type(shape::type_t type)
     MIGRAPHX_THROW("ROCBLAS_GEMM: data type not supported!");
 }
 
+void blas_shape(const shape& s)
+{
+    if(s.lens().size() < 2)
+        return;
+    if(std::none_of(s.strides().end() - 2, s.strides().end(), [&](auto i) { return i == 1; }))
```

Comment:
Should be lengths instead of strides to be collapsible ? I think both works.

Reply:
This isn't checking the batch dimensions. Blas needs one of the strides of the matrix dimension(ie either the row or column) to be `1`, so this checking for that.